bin/gptquery.js

```diff
@@ -0,0 +1,139 @@
+// NOTE: based on python version: https://gist.github.com/pahaz/17c66fd6d75b74ff307aca2b6bf942f3
+const crypto = require('crypto')
+const fs = require('fs')
+const path = require('path')
+
+const fetch = require('node-fetch') // Import fetch from node-fetch
+
+const DEFAULT_OUTPUT_DIR = 'out'
+const GPT_KEY_FILE = path.resolve(process.env.HOME, '.openai.key')
+const GPT_CACHE_DIR = path.resolve(process.env.HOME, '.openai.cache')
+
+function cleanAndTrimText (text) {
+    return text.replace(/[\n\r\t\f\v]/g, ' ').split(/\s+/).join(' ').slice(0, 50).trim()
+}
+
+function loadApiKey () {
+    if (fs.existsSync(GPT_KEY_FILE)) {
+        return fs.readFileSync(GPT_KEY_FILE, 'utf-8').trim()
+    } else if (process.env.OPENAI_API_KEY) {
+        return process.env.OPENAI_API_KEY.trim()
+    } else {
+        throw new Error('API key file not found and OPENAI_API_KEY environment variable is not set.')
```

Comment:
Not sure what I should do for it to work... Should I just set OPENAI_API_KEY or I also need to obtain API key file somewhere?